src/toolkits/drawing_classifier/drawing_classifier.hpp

```diff
@@ -261,11 +260,16 @@ class EXPORT drawing_classifier : public ml_model_base {
     return variant_get_value<T>(get_state().at(key));
   }
 
-  /**
-   * for test svae & load purpose. well, nn_spec_ should not be exposed
-   * to user. I have to do this hack for test purpose
-   */
-  friend void init_dc_model_spec_for_test(drawing_classifier& dc);
+  /* mock test only ; not intended for inheritance */
+  inline void init_model_spec() { nn_spec_ = init_model(); }
```

Comment:
Why can't we just use the protected `drawing_classifier` constructor provided above?

Reply:
right. But I have to provide other parameters.

Besides that, `init_model` is a virtual function, call virtual function during construction is not a good practice.

Reply:
We shouldn't call `init_model` in the constructor, but the constructor can simply accept the intended values for the data members. Please see the OD tests for an example.